wagtail/admin/views/reports/page_types_usage.py

```diff
@@ -0,0 +1,175 @@
+from wagtail.models import ContentType, Page, Site, get_page_models
+from wagtail.permission_policies.pages import PagePermissionPolicy
+from wagtail.users.utils import get_deleted_user_display_name
+
+
+def _get_locale_choices():
+    choices = [  # noqa: C416
+        (language_code, display_name)
+        for language_code, display_name in get_content_languages().items()
+    ]
+    return choices
+
+
+def _get_site_choices():
+    """Tuples of (site root page path, site display name) for all sites in project."""
+    choices = [(site.root_page.path, str(site)) for site in Site.objects.all()]
+    return choices
+
+
+def _annotate_last_edit_info(queryset, language_code, site_root_path):
+    latest_edited_page_filter_kwargs = {}
+    page_count_filter_kwargs = {}
+    if language_code:
+        latest_edited_page_filter_kwargs["locale__language_code"] = language_code
+        page_count_filter_kwargs["pages__locale__language_code"] = language_code
+    if site_root_path:
+        latest_edited_page_filter_kwargs["path__startswith"] = site_root_path
+        page_count_filter_kwargs["pages__path__startswith"] = site_root_path
+
+    latest_edited_page = Page.objects.filter(
+        content_type=OuterRef("pk"), **latest_edited_page_filter_kwargs
+    ).order_by(F("latest_revision_created_at").desc(nulls_last=True), "title", "-pk")
+
+    queryset = queryset.annotate(
+        count=Count("pages", filter=Q(**page_count_filter_kwargs)),
+        last_edited_page_id=Subquery(latest_edited_page.values("pk")[:1]),
+        last_edited_page_owner_id=Subquery(latest_edited_page.values("owner__pk")[:1]),
+    )
+
+    return queryset
+
+
+class LocaleFilter(django_filters.ChoiceFilter):
+    def filter(self, qs, language_code):
+        if language_code:
+            return qs.filter(pages__locale__language_code=language_code)
+        return qs
+
+
+class SiteFilter(django_filters.ChoiceFilter):
+    def filter(self, qs, site_root_path):
+        # Value passed will be the site root page path
+        # To check if a page is in a site, we check if the page path starts with the
+        # site's root page path
+        if site_root_path:
+            return qs.filter(pages__path__startswith=site_root_path)
+        return qs
+
+
+class PageTypesUsageReportFilterSet(WagtailFilterSet):
+    page_locale = LocaleFilter(
+        label=_("Locale"),
+        choices=_get_locale_choices,
+        empty_label=None,
+        null_label=_("All"),
+        null_value=None,
+    )
+    site = SiteFilter(
+        label=_("Site"),
+        choices=_get_site_choices,
+        empty_label=None,
+        null_label=_("All"),
+        null_value=None,
+    )
+
+    class Meta:
+        model = ContentType
+        fields = ["page_locale", "site"]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.sites = {
+            site.pk: site for site in Site.objects.all().prefetch_related("root_page")
```

Comment:
I don't know if this is still true, but it used to be that `prefetch_related()` always made an extra query, whereas `select_related()` would do it in the same one:
```suggestion
            site.pk: site for site in Site.objects.all().select_related("root_page")
```

Reply:
Yep, I went with `select_related`select_related